server.go

```diff
@@ -17,18 +18,41 @@ import (
 
 // The server handles creating connections and listening on various ports.
 type Server struct {
-	n              *internal.Node
-	listeners      map[string]*internal.SSHListener
-	listen_address string
+	n         *internal.Node
+	listeners map[string]*internal.SSHListener
+	// This contains the addresses of the nodes that we are currently connecting
+	// to, so we don't try to connect to them twice at the same time.
+	currently_connecting map[string]bool
```

Comment:
So the internal node structure already has a list of connections. Can we have it be smarter about removing duplicate connections? It seems like that's likely the right spot to do it. Also adding a polling methods so we can not make SSH connections if we already have an active one sounds like a good idea.